pkg/pebinaryexecutor/pebinary.go

```diff
@@ -0,0 +1,268 @@
+// +build windows
+
+package pebinaryexecutor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/Microsoft/hcsshim/hcn"
+	"github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io"
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	daemonconfig "github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/daemons/executor"
+	"github.com/rancher/rke2/pkg/bootstrap"
+	"github.com/rancher/rke2/pkg/images"
+	win "github.com/rancher/rke2/pkg/windows"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	ssldirs = []string{
+		"/etc/ssl/certs",
+		"/etc/pki/tls/certs",
+		"/etc/ca-certificates",
+		"/usr/local/share/ca-certificates",
+		"/usr/share/ca-certificates",
+	}
+)
+
+type PEBinary struct {
```

Comment:
for parity with `podexecutor.StaticPodConfig`
```suggestion
type PEBinaryConfig struct {
```